lib/tests/streamlit/web/cli_test.py

```diff
@@ -404,6 +404,38 @@ def test_activate_command(self):
             self.runner.invoke(cli, ["activate"])
             mock_credential.activate.assert_called()
 
+    @tempdir()
+    def test_email_send(self, temp_dir):
+        """Test that saving a new Credential sends an email"""
+
+        with requests_mock.mock() as m:
+            m.post("https://api.segment.io/v1/i", status_code=200)
+            creds: Credentials = Credentials.get_current()  # type: ignore
+            creds._conf_file = str(Path(temp_dir.path, "config.toml"))
+            creds.activation = _verify_email("email@test.com")
+            creds.save()
+            last_request = m.request_history[-1]
+            assert last_request.method == "POST"
+            assert last_request.url == "https://api.segment.io/v1/i"
+            assert "'userId': 'email@test.com'" in last_request.text
+
+    # @patch("streamlit.runtime.credentials.LOGGER")
+    @tempdir()
+    def test_email_send_exception_handling(self, temp_dir):
+        """Test that saving a new Credential sends an email"""
```

Comment:
Nit: "test that saving a new Credential catches and logs failures from the segment endpoint"